admin/controllers/service-manager.js

```diff
@@ -13,374 +13,343 @@ const sortingAttributesService = require('../services/sorting-attributes.service
+    res.locals.pageTitle = 'Manage check windows'
+    req.breadcrumbs(res.locals.pageTitle)
+    let checkWindows
+    // Sorting
+    const sortingOptions = [
+      { 'key': 'checkWindowName', 'value': 'asc' },
+      { 'key': 'adminStartDate', 'value': 'asc' },
+      { 'key': 'checkStartDate', 'value': 'asc' }
+    ]
+    const sortField = req.params.sortField === undefined ? 'checkWindowName' : req.params.sortField
+    const sortDirection = req.params.sortDirection === undefined ? 'asc' : req.params.sortDirection
+    const { htmlSortDirection, arrowSortDirection } = sortingAttributesService.getAttributes(sortingOptions, sortField, sortDirection)
+
+    // Get all check windows
+    try {
+      checkWindows = await checkWindowService.getAllCheckWindows(sortField, sortDirection)
+    } catch (error) {
+      return next(error)
+    }
 
-  if (validationError.hasError()) {
+    res.render('service-manager/check-windows', {
+      breadcrumbs: req.breadcrumbs(),
+      checkWindowList: checkWindows,
+      htmlSortDirection,
+      arrowSortDirection
+    })
+  },
+
+  /**
+   * Add window check form.
+   * @param req
+   * @param res
+   * @param next
+   * @returns {Promise.<void>}
+   */
+  checkWindowsForm: async (req, res, next) => {
+    let error = ''
+    let errorMessage = ''
+    let checkWindowData = ''
+    let successfulPost = false
+    let actionName = (req.params.action === 'add' && req.params.id === undefined) ? 'Create' : 'Edit'
+    let urlActionName = req.params.action
+    let currentYear = moment.utc(Date.now()).format('YYYY')
+    let adminIsDisabled = 0
+    let checkStartIsDisabled = 0
+
+    req.breadcrumbs('Manage check windows', '/service-manager/check-windows')
     res.locals.pageTitle = actionName + ' check window'
     req.breadcrumbs(res.locals.pageTitle)
 
-    if (!req.body['adminStartDay'] && !req.body['adminStartMonth'] && !req.body['adminStartYear'] && req.body['existingAdminStartDate'] && req.body['adminIsDisabled'] === '1') {
-      req.body.adminStartDay = moment(req.body['existingAdminStartDate']).format('D')
-      req.body.adminStartMonth = moment(req.body['existingAdminStartDate']).format('MM')
-      req.body.adminStartYear = moment(req.body['existingAdminStartDate']).format('YYYY')
-    }
-
-    if (!req.body['checkStartDay'] && !req.body['checkStartMonth'] && !req.body['checkStartYear'] && req.body['existingCheckStartDate'] && req.body['checkStartIsDisabled'] === '1') {
-      req.body.checkStartDay = moment(req.body['existingCheckStartDate']).format('D')
-      req.body.checkStartMonth = moment(req.body['existingCheckStartDate']).format('MM')
-      req.body.checkStartYear = moment(req.body['existingCheckStartDate']).format('YYYY')
+    if (req.params.id !== undefined) {
+      try {
+        checkWindowData = await checkWindowDataService.sqlFindOneById(req.params.id)
+
+        const adminStartDate = moment(checkWindowData.adminStartDate, 'D MM YYYY').format('YYYY-MM-D')
```

Comment:
should all this moment parsing be in the service?

Reply:
@GuyHarwood this is relevant to add/editing check window which is going to be a separate branch and PR. As described this appears as changed because the controller declaration changed.